src/main/java/seedu/bookmark/ui/DetailedBookCard.java

```diff
@@ -20,5 +25,14 @@ public DetailedBookCard(Book book, int displayedIndex) {
     @Override
     protected void initialize(Book book, int displayedIndex) {
         super.initialize(book, displayedIndex);
+        int percentageCompleted = calculateCompletion(book);
+        String progressDisplay = "Progress: " + String.valueOf(percentageCompleted) + "%";
+        percentageCompletion.setText(progressDisplay);
+    }
+
+    private int calculateCompletion(Book book) {
+        int pagesRead = book.getPagesRead();
+        int totalPages = book.getTotalPagesNumber();
+        return (int) (((float) pagesRead / totalPages) * 100);
```

Comment:
Maybe it will be better to immediately cast the float type upon instantiation instead of typecasting at Ln36? 

Else, LGTM!

Reply:
Good suggestion! I've made the necessary changes